### ISSUE TITLE
[layers] Add a 1-layer linear (kernel-feature) attention block

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#5665]
## Summary

Add `LinearAttention`, a standalone `Module`-conforming linear-attention
primitive to `src/odyssey/core/layers/`. It replaces the softmax kernel of
scaled dot-product attention with the **linearized (kernel-feature) attention**
of Katharopoulos et al. 2020 ("Transformers are RNNs: Fast Autoregressive
Transformers with Linear Attention", ICML 2020, arXiv:2006.16236, §3.2):

```
attention(Q, K, V)_i = ( φ(Q_i)ᵀ Σ_j φ(K_j) V_jᵀ ) / ( φ(Q_i)ᵀ Σ_j φ(K_j) )   (eq. 5)
φ(x) = elu(x) + 1                                                              (eq. 7)
```

This is the sibling of the softmax `MultiHeadAttention` block
(`core/layers/attention.mojo`): same QKV-projection / head-split / head-merge
conventions, but the S×S score matrix is never materialized — the key–value
summary `φ(K)ᵀV` and key summary `Σ φ(K)` are each computed ONCE per head, so
cost is O(S·d_k²) in sequence length rather than O(S²·d_k). This is the
associativity trick that gives linear attention its O(N) complexity.

## Scope

- **Non-causal (full-sequence) form** — every query attends to every key
  (§3.2 eq. 5). The paper's causal/autoregressive recurrence (§3.3, where the
  summaries accumulate as running RNN-like states) is deliberately out of scope
  for this bare primitive.
- **Batch-first** `[batch, seq, dim]`, full-sequence `forward()`.
- **Self-contained forward** — does NOT call the functional attention core
  (`core/attention.mojo`), which is broken at HEAD (Odyssey#5648). Linear
  attention has no softmax to reuse there anyway; the kernelized path is
  independent.
- **No BatchNorm / LayerNorm / residual** — bare primitive, measured in
  isolation; composition is external.

## Key properties (tested)

- **O(N) ≡ O(N²) equivalence.** The efficient order φ(Q)(φ(K)ᵀV) is asserted
  equal to the naive order (φ(Q)φ(K)ᵀ)V to 1e-10 (f64) on a small config — a
  regression tripwire for the associativity refactor.
- **Denominator guard.** φ(x) = elu(x) + 1 > 0 for all x, so the normalizer
  Σ φ(K) is strictly positive; a small `eps` (default 1e-6) is added before the
  division to guard the degenerate all-near-zero case. Documented in the
  docstring and MIRRORED in the parity reference so parity holds to 1e-5.

## Files

| File | Purpose |
| --- | --- |
| `src/odyssey/core/layers/linear_attention.mojo` | `LinearAttention` struct (Module-conforming layer) |
| `tests/odyssey/core/layers/test_linear_attention.mojo` | 10 unit tests (shape, validation, 2× parity, O(N)-vs-O(N²), positivity, export) |
| `tests/odyssey/core/layers/parity_refs/linear_attention_parity_reference.py` | torch generator (ramp-seeded, f64, prints JSON) |
| `tests/odyssey/core/layers/parity_refs/linear_attention_parity_reference.json` | committed fixture |
| `src/odyssey/core/layers/__init__.mojo` | one export line + one docstring bullet |
| `scripts/run_primitive_test.sh` | `TEST[linear_attention]=…` registration |
| `README.md` | one primitive-table row |

## Acceptance criteria

- [ ] `LinearAttention[dtype]` conforms to `Module` (`forward`, `parameters`,
      `train`, `eval`); float32 default, float64 for parity.
- [ ] `parameters()` returns 8 tensors (q/k/v/out weight+bias) in projection
      order.
- [ ] PyTorch parity to 1e-5 on single-head and multi-head configs.
- [ ] O(N)-vs-O(N²) associativity equivalence test.
- [ ] Denominator-positivity test.
- [ ] `--Werror`-clean compile; `mojo format`, ruff, markdownlint, pre-commit
      all pass.

## Reference

Katharopoulos, A., Vyas, A., Pappas, N., & Fleuret, F. (2020). Transformers are
RNNs: Fast Autoregressive Transformers with Linear Attention. ICML 2020.
arXiv:2006.16236, §3.2 "Linearized Attention" (eq. 5, eq. 7).

Tracking: mvillmow/Random#60 ([ARCH-10] 1-layer linear attention block).
